internal/testutil/asserter/assert.go

```diff
@@ -175,32 +150,44 @@ func entitiesFor(entities []*integration.Entity, pseudotype string) []*integrati
 	return appropriateEntities
 }
 
-// entityHas returns true if supplied entity has metric m with type _similar_ to mType, false otherwise.
-func entityHas(e *integration.Entity, m string, mType metric.SourceType) bool {
+// entityMetric is a helper function that returns the first metric from an entity that matches the given name.
+func entityMetric(e *integration.Entity, m string) interface{} {
+	for _, ms := range e.Metrics {
+		entityMetric, found := ms.Metrics[m]
+		if found {
+			return entityMetric
+		}
+	}
+
+	return nil
+}
+
+// EntityHas returns true if the specified entity has a metric of name m equal to value.
+func EntityHas(e *integration.Entity, m string, value interface{}) bool {
```

Comment:
Whats about `entityMetricWithWildcard`? Just a suggestion, feel free to ignore it :)